src/common/inspector/types.js

```diff
@@ -0,0 +1,43 @@
+(function() {
+  'use strict';
+
+  function isNativeType(typeName) {
+    var nativeTypes = ['string', 'boolean', 'number', 'integer', 'object'];
+    return nativeTypes.indexOf(typeName) !== -1;
+  }
+
+  function getType(typeName, types) {
+    typeName = typeName.replace('[]', '');
+    var existingType = types.find(function (aType) {
+      return aType[typeName];
+    });
+    return existingType ? existingType[typeName] : existingType;
+  }
+
+  function getSuperTypesProperties(properties, typeName, types) {
+    if (!isNativeType(typeName)) {
+      var superType = getType(typeName, types);
+
+      if (superType) {
+        properties = Object.assign({}, superType.properties, properties);
+        return getSuperTypesProperties(properties, superType.type[0], types);
+      }
+    } else {
+      return properties;
+    }
+  }
+
+  function mergeType(type, types) {
+    if (!isNativeType(type.type[0])) {
+      var resultingType = angular.copy(type);
+      resultingType.properties = getSuperTypesProperties(type.properties, type.type[0], types);
```

Comment:
yes, the order of the properties is done inside `getSuperTypesProperties` :
`properties = Object.assign({}, superType.properties, properties);`

That way, super type properties always come first and property overrides in child types work